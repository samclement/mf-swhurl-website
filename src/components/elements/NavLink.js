import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

const StyledLink = styled.a`
  color: black;
  padding: 5px 10px;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 12px;
  font-weight: bold;
  &:hover {
    color: #f00;
  }
  &:visited {
    color: black;
  }
  @media only screen and (max-width: 480px) {
    padding: 5px;
  }
`

export default ({ as, href, label }) => (
  <Link as={as} href={href} passHref>
    <StyledLink>{label}</StyledLink>
  </Link>
)
