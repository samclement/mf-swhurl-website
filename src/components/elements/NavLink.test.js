import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import NavLink from './NavLink.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {}, prefetch: () => {} }
Router.router = mockedRouter

describe('With Snapshot Testing', () => {
  it('NavLink shows "About" with href "/about"', () => {
    const { asFragment } = render(<NavLink href="/about" name="About" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
