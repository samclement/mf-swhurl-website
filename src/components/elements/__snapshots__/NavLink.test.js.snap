// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing NavLink shows "About" with href "/about" 1`] = `
<DocumentFragment>
  .c0 {
  color: black;
  padding: 5px 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 12px;
  font-weight: bold;
}

.c0:hover {
  color: #f00;
}

.c0:visited {
  color: black;
}

@media only screen and (max-width:480px) {
  .c0 {
    padding: 5px;
  }
}

<a
    class="c0"
    href="/about"
  />
</DocumentFragment>
`;
